lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMPass.cpp

```diff
@@ -309,6 +310,54 @@ class ConvertTritonGPUToLLVM
     });
   }
 
+  // Replace pattern extf(trans(cvt(x))) with trans(extf(cvt(x)))
+  //
+  // Currently TransOp will always use shared encoding in ttgir. If input is not
+  // in shared encoding convert_layout will be inserted. In flash attention,
+  // this causes extf operation (that comes after trans) to use shared
+  // encoding. Triton currently doesn't support code generation for elmtwise ops
+  // in shared encoding. To work around this issue, order of these instructions
+  // is inverted (since it doesn't affect the result), thus ensuring extf is not
+  // using shared encoding.
```

Comment:
Ideally, we should probably either be able to have elmtwise codegen work for shared encoding or at least
be able to convert back to blocked layout (currently shared->blocked is not supported, nor it can be easily decomposed). This should be temporary fix that enables further investigation of flash attention issues.